### PR TITLE
Use Opaque to represent empty Structs, update tests (#142, #143)

### PR DESCRIPTION
**Pull request by mannprerak2**

### CODE REVIEW

File: example/c_json/pubspec.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@
 name: c_json_example
 
 environment:
-  sdk: '>=2.12.0-198.0.dev <3.0.0'
+  sdk: '>=2.12.0-230.0.dev <3.0.0'
```
Comment (dcharkes) [inline example/c_json/pubspec.yaml:8]: Should this also be 237? (Also in other places.)

**Review by dcharkes — APPROVED** [review]
Already reviewed in #142 and #143.

File: example/c_json/pubspec.yaml (hunk 1 above)
Comment (mannprerak2) [reply example/c_json/pubspec.yaml]: Done.

Comment (mannprerak2) [reply example/c_json/pubspec.yaml]: This hint for empty structs was added in 230 so I don't think this should of any issue.

**Pull request merged**